source/common/config/decoded_resource_impl.h

```diff
@@ -91,6 +98,7 @@ class DecodedResourceImpl : public DecodedResource {
   const std::string version_;
   // Per resource TTL.
   const absl::optional<std::chrono::milliseconds> ttl_;
+  const OptRef<const envoy::config::core::v3::Metadata> metadata_;
```

Comment:
Can you add a comment about the lifetime of the original metadata, and why it is ok to keep a ref here.